be/src/runtime/user_function_cache.cpp

```diff
@@ -216,10 +216,12 @@ Status UserFunctionCache::_get_cache_entry(int64_t fid, const std::string& url,
     std::string file_name = _get_file_name_from_url(url);
     {
         std::lock_guard<std::mutex> l(_cache_lock);
-        auto it = _entry_map.find(fid);
-        if (it != _entry_map.end()) {
-            entry = it->second;
-        } else {
+        for (const auto& item : _entry_map) {
+            if (item.second->checksum == checksum) {
```

Comment:
seems use func id or checksum , it's also need restart.
maybe we should remove all cache jar under lib/udf folder when start BE.

Reply:
If you locate the jar based on the `checksum` parameter, there is no need to restart Be. This `checksum` is generated by the Fe end parsing the new jar package when creating the udf function.
refer to：`org.apache.doris.analysis.CreateFunctionStmt.computeObjectChecksum()`